server/text_generation_server/models/custom_modeling/flash_llama_modeling.py

```diff
@@ -368,6 +373,23 @@ def forward(
         cos, sin = self.layers[0].self_attn.rotary_emb.get_cos_sin(
             position_ids, max_s, hidden_states.dtype
         )
+        if cu_seqlen_prefill is None and FLASH_DECODING:
```

Comment:
Maybe we could do this in the `paged_attention` function? Then it has a non-ambiguous signature and we don't have to add this to all the models.

Reply:
The problem is that the tensor creation adds too much overhead.

I did it that way initially  and the performance were worse than raw paged just because of that.

Reply:
We could also maybe to it all the way up in `flash_causal_lm`. That was my next best idea (but I don't like obfuscating tensor content since then the tensors might be either `cu_seqlen_q` and cu_seqlen_k` or `None` and `input_lengths` (we could dataclass stuff and all sorts of shenanigans, still obfuscation I feel.

Given the totally optional nature of flash decoding for now, I'm ok if this lives into this particular modeling code while we test, and either rollback or finish the work and put everything into causal_lm once there's only 1 format (biggest drawback will be AMD and intel which do not support FA2 with paged afaik)